### PR TITLE
SAK-49581 T&Q: Added sharing permissions to QPs

### PR DESCRIPTION
**Pull request by mpellicer**

### CODE REVIEW

File: samigo/samigo-app/src/webapp/jsf/questionpool/sharePool.jsp [hunk 1]
```diff
@@ -179,12 +195,12 @@ function checkUpdate()
 <br/>
 
 <div class="tier1">
-<h4><h:outputText value="#{questionPoolMessages.members_without_access} #{questionpoolshare.questionPoolName}"/></h4>
+<h4><h:outputText value="#{questionPoolMessages.members_without_access} \"#{questionpoolshare.questionPoolName}\" "/></h4>
```
[the remark this answers is not in the document]
Comment (mpellicer) [reply samigo/samigo-app/src/webapp/jsf/questionpool/sharePool.jsp:198]: False positive

File: samigo/samigo-app/src/webapp/jsf/questionpool/accesstype_whatsThis.jsp [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+-->
+
+
+<f:view>
+    <html xmlns="http://www.w3.org/1999/xhtml">
+        <head><%= request.getAttribute("html.head") %>
+            <title><h:outputText value="#{questionPoolMessages.whatsThis_body}" escape="false"/></title>
+        </head>
+        <body onload="<%= request.getAttribute("html.body.onload") %>">
+            <h2 class="western">
```
[the remark this answers is not in the document]
Comment (mpellicer) [reply samigo/samigo-app/src/webapp/jsf/questionpool/accesstype_whatsThis.jsp:40]: false positive

File: samigo/samigo-app/src/webapp/jsf/questionpool/previewQuestion.jsp [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+<%@ include file="/js/samigotree.js" %>
+//-->
+</script>
+      </head>
+<body onload="<%= request.getAttribute("html.body.onload") %>">
+<!-- content... -->
+ <div class="portletBody">
+<h:form id="previewQuestion">
+<h:messages infoClass="validation" warnClass="validation" errorClass="validation" fatalClass="validation"/>
+<h3><h:outputText value="#{questionPoolMessages.t_previewQuestion}"/></h3>
```
[the remark this answers is not in the document]
Comment (mpellicer) [reply samigo/samigo-app/src/webapp/jsf/questionpool/previewQuestion.jsp:26]: false positive

File: samigo/samigo-app/src/webapp/jsf/questionpool/sharePool.jsp [hunk 4]
```diff
@@ -91,7 +91,7 @@ function checkUpdate()
 <h:messages styleClass="sak-banner-error" rendered="#{! empty facesContext.maximumSeverity}" layout="table"/>
  
 <div class="tier1">
-<h4><h:outputText value="#{questionPoolMessages.members_with_access} #{questionpoolshare.questionPoolName}"/></h4>
+<h4><h:outputText value="#{questionPoolMessages.members_with_access} \"#{questionpoolshare.questionPoolName}\" "/></h4>
```
[the remark this answers is not in the document]
Comment (mpellicer) [reply samigo/samigo-app/src/webapp/jsf/questionpool/sharePool.jsp:94]: false positive

**Review by bgarciaentornos — CHANGES_REQUESTED** [review]
This is a tough one to review! Found a couple things I'd like to know more about but nothing outstanding...

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/jsf/converter/AccessTypeConverter.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        String result = "";
+        String value = "";
+        if ("java.lang.String".equals(arg2.getClass().getName())) {
+            value = (String) arg2;
+        } else if ("java.lang.Long".equals(arg2.getClass().getName())) {
+            value = ((Long) arg2).toString();
+        }
+
+        switch(value) {
+            case "31":
```
[the remark this answers is not in the document]
Comment (mpellicer) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/jsf/converter/AccessTypeConverter.java:43]: Okay

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/questionpool/QuestionPoolBean.java [hunk 6]
```diff
@@ -2341,12 +2328,12 @@ private void setPoolInfo(QuestionPoolDataBean pool) {
 	pool.setKeywords(keyField);
   }
   
-  	public void setQpDataModelByLevel() {
+  	public void setQpDataModelByLevel(String sortProperty, boolean sortAscending) {
```
[the remark this answers is not in the document]
Comment (mpellicer) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/questionpool/QuestionPoolBean.java:2331]: Probably copy/pasta from 13 years ago, you are right

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/questionpool/QuestionPoolBean.java [hunk 7]
```diff
@@ -2359,42 +2346,16 @@ public void setQpDataModelByLevel(Long poolId) {
...
 
 		Collection objects = tree.getSortedObjects();
 		ListDataModel model = new ListDataModel((List) objects);
-		QuestionPoolDataModel qpDataModel = new QuestionPoolDataModel(tree,
-				model);
+		QuestionPoolDataModel qpDataModel = new QuestionPoolDataModel(tree, model);
 		this.qpDataModel = qpDataModel;
 	}
 
-  	public void setQpDataModelByLevelCopy(String sortProperty, boolean sortAscending) {
```
[the remark this answers is not in the document]
Comment (mpellicer) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/questionpool/QuestionPoolBean.java:2373]: It is redundant I think

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/questionpool/QuestionPoolBean.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (mpellicer) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/questionpool/QuestionPoolBean.java]: Done, I've reverted these changes, the bean attributes are being used.

**Comment by mpellicer** [comment]
I'll do another round of testing focused on sorting QPs by columns.

**Comment by mpellicer** [comment]
> I'll do another round of testing focused on sorting QPs by columns.

@bgarciaentornos Sorting still works after reverting the changes about the unused params.

zz

**Review by bgarciaentornos — APPROVED** [review]
Nice, thanks!

**Comment by mpellicer** [comment]
Thank you very much for the review

**Pull request merged**